### DISCUSSION TITLE
Access-control-allow-origin header settings

### DISCUSSION THREAD

**Discussion by mFahadYounas** [pocketbase/pocketbase#3946] [Q&A] [answered]
For learning purposes, I am trying to make a react app that requires authentication and I am using the REST API of Pocket Base for authentication. I want to know, how can I set 'Access-Control-Allowed-Credentials' to be true when I serve Pocket Base.
Thank you in advance.

**Comment by pnmcosta** [comment]
AFAIK the only way would be by https://pocketbase.io/docs/use-as-framework/ - at least that's how I use and am able to set my own https://github.com/labstack/echo route headers.

But I'm 99% sure you should be able to set a middleware to ensure the header is also set for all pocketbase api routes too.

**Reply by pnmcosta** [reply, marked as answer]
Also see https://pocketbase.io/docs/go-event-hooks/#onbeforeserve which has an example of setting a custom route, could be a good starting point for you.
